src/main/java/Task.java

```diff
@@ -0,0 +1,36 @@
+public class Task {
```

Comment:
Any reason why you did not make the Task class abstract since you did not directly instantiate a Task object anyway?